### DISCUSSION TITLE
What settings do we need at the module level?

### DISCUSSION THREAD

**Discussion by jgostick** [PMEAL/porespy#365] [General]
I have added a new dataclass to the module, so that ``porespy.settings`` can store various properties for controlling the behavior of the package.  So far I have only added one setting: ``show_progress``.  When ``True``, the default, ``tqdm`` outputs its messages, while setting it to ``False`` turns off the output.

I am wondering, what other setting might be useful?  I am opening this discussion as a place to brainstorm ideas.  What behaviors of the package to we wish we could control?

**Comment by ma-sadeghi** [comment]
Here are some thoughts:
- Matplotlib
    - stylesheet
    - figure size
    - dpi
- Paraview path maybe? (until we list paraview as package dependency)

**Comment by jgostick** [comment]
I am adding an argument called ``smooth`` to several functions, which controls whether or not inserted spheres should be found with ``dt < r`` (smooth) or ``dt <= r`` (not smooth).  This *might* be something to put at the settings level, but not sure.  It would certainly clean-up and simplify the relevant function calls and docstrings.
